File: ios/FreelanceKar/Views/Screens/GigsScreen.swift
import SwiftUI

struct GigsScreen: View {
    @EnvironmentObject var store: AppStore

    var body: some View {
        let s = store.state
        ScreenShell {
            VStack(alignment: .leading, spacing: 0) {
                Text("Gigs").font(.fkH1).padding(.bottom, 6)
                Text("\(s.fGigs.count) active · \(s.quotesSent) quotes awaiting reply")
                    .font(.system(size: 12.5)).foregroundStyle(FK.text50).padding(.bottom, 16)

                if s.fGigs.isEmpty {
                    VStack(spacing: 0) {
                        Text("◫").font(.system(size: 22)).foregroundStyle(FK.text28)
                        Text("No gigs running. Quote on requests from the dashboard — accepted quotes turn into gigs here.")
                            .font(.fkBody).foregroundStyle(FK.text45).multilineTextAlignment(.center).padding(.top, 12)
                        Button { store.go(.fdash) } label: {
                            Text("See new requests").font(.system(size: 12.5, weight: .semibold)).foregroundStyle(FK.onLime)
                                .padding(.horizontal, 16).padding(.vertical, 12)
                        }
                        .background(FK.lime)
                        .clipShape(RoundedRectangle(cornerRadius: 11, style: .continuous))
                        .padding(.top, 16)
                    }
                    .frame(maxWidth: .infinity)
                    .padding(.horizontal, 22).padding(.top, 40)
                } else {
                    VStack(spacing: 10) {
                        ForEach(s.fGigs) { g in
                            let done = g.status == .DELIVERED
                            VStack(spacing: 0) {
                                HStack(spacing: 11) {
                                    Stripe(cornerRadius: 11).frame(width: 36, height: 36)
                                    VStack(alignment: .leading, spacing: 2) {
                                        Text(g.client).font(.fkCardTitle)
                                        Text(g.service).font(.system(size: 11)).foregroundStyle(FK.text50)
                                    }
                                    Spacer()
                                    Text(g.amount).font(.system(size: 13, weight: .semibold))
                                }
                                HStack {
                                    Text(g.status.rawValue).font(.fkMonoSm).foregroundStyle(done ? FK.text45 : FK.lime)
                                    Spacer()
                                    Button { store.markGigDelivered(g.id) } label: {
                                        Text(done ? "Awaiting approval" : "Mark delivered")
                                            .font(.system(size: 11.5, weight: .semibold))
                                            .foregroundStyle(done ? FK.text60 : FK.onLime)
                                            .padding(.horizontal, 13).padding(.vertical, 9)
                                    }
                                    .background(done ? Color.clear : FK.lime)
                                    .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
                                    .overlay(RoundedRectangle(cornerRadius: 10).stroke(done ? FK.line14 : .clear, lineWidth: 1))
                                }
                                .padding(.top, 11)
                                .overlay(Rectangle().frame(height: 1).foregroundStyle(FK.line07), alignment: .top)
                                .padding(.top, 12)
                            }
                            .padding(14)
                            .background(FK.card)
                            .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                            .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line08, lineWidth: 1))
                        }
                    }
                }
            }
            .padding(.horizontal, 20).padding(.top, 18).padding(.bottom, 26)
        }
    }
}

File: android/app/src/main/java/com/freelancekar/app/ui/screens/GigsScreen.kt
package com.freelancekar.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.AppState
import com.freelancekar.app.data.Screen
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.Stripe
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

@Composable
fun GigsScreen(state: AppState, vm: AppViewModel) {
    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(horizontal = 20.dp).padding(top = 18.dp, bottom = 26.dp)) {
        Text("Gigs", style = FkType.h1)
        Spacer(Modifier.height(6.dp))
        Text("${state.fGigs.size} active · ${state.quotesSent} quotes awaiting reply", style = FkType.meta.copy(color = FkColor.text50))
        Spacer(Modifier.height(16.dp))

        if (state.fGigs.isEmpty()) {
            Column(Modifier.fillMaxWidth().padding(top = 40.dp), horizontalAlignment = Alignment.CenterHorizontally) {
                Text("◫", style = FkType.h2.copy(color = FkColor.text28))
                Spacer(Modifier.height(12.dp))
                Text(
                    "No gigs running. Quote on requests from the dashboard — accepted quotes turn into gigs here.",
                    style = FkType.body.copy(color = FkColor.text45),
                    textAlign = TextAlign.Center,
                )
                Spacer(Modifier.height(16.dp))
                androidx.compose.foundation.layout.Box(
                    Modifier
                        .clip(RoundedCornerShape(11.dp))
                        .background(FkColor.lime)
                        .clickable { vm.go(Screen.FDASH) }
                        .padding(horizontal = 16.dp, vertical = 12.dp),
                ) {
                    Text("See new requests", style = FkType.meta.copy(color = FkColor.onLime, fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                }
            }
        } else {
            Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                state.fGigs.forEach { g ->
                    val done = g.status == "DELIVERED"
                    Column(
                        Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(15.dp))
                            .background(FkColor.card)
                            .border(1.dp, FkColor.line08, RoundedCornerShape(15.dp))
                            .padding(14.dp),
                    ) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Stripe(Modifier.size(36.dp).clip(RoundedCornerShape(11.dp)))
                            Spacer(Modifier.width(11.dp))
                            Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                                Text(g.client, style = FkType.cardTitle)
                                Text(g.service, style = FkType.meta.copy(color = FkColor.text50))
                            }
                            Text(g.amount, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                        }
                        Row(
                            Modifier.fillMaxWidth().padding(top = 11.dp).border(0.dp, FkColor.line07).padding(top = 0.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically,
                        ) {
                            Text(g.status, style = FkType.monoSm.copy(color = if (done) FkColor.text45 else FkColor.lime))
                            if (done) {
                                Row(
                                    Modifier
                                        .clip(RoundedCornerShape(10.dp))
                                        .border(1.dp, FkColor.line14, RoundedCornerShape(10.dp))
                                        .clickable { vm.markGigDelivered(g.id) }
                                        .padding(horizontal = 13.dp, vertical = 9.dp),
                                ) { Text("Awaiting approval", style = FkType.meta.copy(color = FkColor.text60, fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold)) }
                            } else {
                                Row(
                                    Modifier
                                        .clip(RoundedCornerShape(10.dp))
                                        .background(FkColor.lime)
                                        .clickable { vm.markGigDelivered(g.id) }
                                        .padding(horizontal = 13.dp, vertical = 9.dp),
                                ) { Text("Mark delivered", style = FkType.meta.copy(color = FkColor.onLime, fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold)) }
                            }
                        }
                    }
                }
            }
        }
    }
}
